alchemist-loading/src/main/kotlin/it/unibo/alchemist/loader/export/CSVExporter.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2010-2021, Danilo Pianini and contributors
+ * listed in the main project's alchemist/build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.loader.export
+
+import com.google.common.base.Charsets
+import it.unibo.alchemist.model.implementations.times.DoubleTime
+import it.unibo.alchemist.model.interfaces.Environment
+import it.unibo.alchemist.model.interfaces.Position
+import it.unibo.alchemist.model.interfaces.Reaction
+import it.unibo.alchemist.model.interfaces.Time
+import java.io.File
+import java.io.PrintStream
+import java.text.SimpleDateFormat
+import java.util.Arrays
+import java.util.Date
+import java.util.Locale
+import java.util.TimeZone
+import kotlin.io.path.Path
+
+/**
+ * Writes on file data provided by a number of {@link Extractor}s. Produces a
+ * CSV with '#' as comment character.e
+ * @param filename the name the file to export data to.
+ * @param interval the sampling time, defaults to [AbstractExporter.DEFAULT_INTERVAL].
+ */
+
+class CSVExporter<T, P : Position<P>> @JvmOverloads constructor(
+    private val filename: String = "",
+    var interval: Double = DEFAULT_INTERVAL,
+    private val appendTime: Boolean = false
+) : AbstractExporter<T, P>(interval) {
+
+    companion object {
+        /**
+         * Character used to separate comments from data on export files.
+         */
+        const val SEPARATOR = "#####################################################################"
+
+        /**
+         * If no path is specified or the input path is wrong, the output file will be placed inside this folder.
+         */
+        const val DEFAULT_PATH = "/build/exports/"
+    }
+    private lateinit var out: PrintStream
```

Comment:
rename this property into something that's not a keyword

Reply:
Done: changed name of property.

Reply:
Done: pushed latest changes now.